docs/requirements.txt

```diff
@@ -7,3 +7,8 @@ opt_einsum>=2.3.2
 pyro-api>=0.1.1
 tqdm>=4.36
 funsor[torch]
+nbformat
+nbsphinx>=0.3.2
+nbstripout
+pypandoc
+ninja
```

Comment:
I worry this will make our readthedocs build slower and flakier, which is a problem because the rtd debugging interface is cumbersome.  Could we instead add these dependencies to the docs stage in ci.yml, so they are triggered in github's easier to debug workflow?